scene/resources/material.cpp

```diff
@@ -324,6 +324,28 @@ void ShaderMaterial::_get_property_list(List<PropertyInfo> *p_list) const {
 					is_uniform_type_compatible = E->get().type == cached.get_type();
 				}
 
+#ifndef DISABLE_DEPRECATED
+				// PackedFloat32Array -> PackedVector4Array conversion.
+				if (!is_uniform_type_compatible && E->get().type == Variant::PACKED_VECTOR4_ARRAY && cached.get_type() == Variant::PACKED_FLOAT32_ARRAY) {
+					PackedVector4Array varray;
+					PackedFloat32Array array = (PackedFloat32Array)cached;
+
+					for (int i = 0; i < array.size(); i += 4) {
+						Vector4 v = Vector4();
+
+						v.x = array[i];
+						v.y = (i + 1 < array.size()) ? array[i + 1] : 0.0f;
+						v.z = (i + 2 < array.size()) ? array[i + 2] : 0.0f;
+						v.w = (i + 3 < array.size()) ? array[i + 3] : 0.0f;
+
+						varray.push_back(v);
+					}
```

Comment:
Can maybe be simplified like this, so we skip the last iteration if it's truncated. It shouldn't happen anyway as that's a degenerate case, so it's fine IMO to cut corners in this case. (Didn't test whether my `i + 3 < array.size()` bound logic works.)
```suggestion
					for (int i = 0; i + 3 < array.size(); i += 4) {
						varray.push_back(Vector4(array[i], array[i + 1], array[i + 2], array[i + 3]);
					}
```

Reply:
Yes, it works.